resources/iam/resource_machine_user.go

```
@@ -0,0 +1,264 @@
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read API call logic
+	client := r.client.Iam
+
+	params := operations.NewListMachineUsersParamsWithContext(ctx)
+	params.WithInput(&models.ListMachineUsersRequest{
+		MachineUserNames: []string{data.Name.ValueString()},
+	})
+
+	responseOk, err := client.Operations.ListMachineUsers(params)
+	if err != nil {
+		utils.AddIamDiagnosticsError(err, &resp.Diagnostics, "read Machine User")
+		if d, ok := err.(*operations.CreateMachineUserDefault); ok && d.GetPayload() != nil && d.GetPayload().Code == "NOT_FOUND" {
+			resp.Diagnostics.AddWarning("Resource not found on provider", "Machine User not found, removing from state.")
+			tflog.Warn(ctx, "Machine User not found, removing from state", map[string]interface{}{
+				"id": data.Id,
+			})
+			resp.State.RemoveResource(ctx)
+		}
+		return
+	}
+
+	if len(responseOk.Payload.MachineUsers) == 1 && responseOk.Payload.MachineUsers[0] != nil {
+		mu := responseOk.Payload.MachineUsers[0]
+		muRespToModel(ctx, mu, &data)
+
+		// Save data into Terraform state
+		resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	} else {
+		if d, ok := err.(*operations.CreateMachineUserDefault); ok && d.GetPayload() != nil && d.GetPayload().Code == "NOT_FOUND" {
+			resp.Diagnostics.AddWarning("Resource not found on provider", "Machine User not found, removing from state.")
+			tflog.Warn(ctx, "Machine User not found, removing from state", map[string]interface{}{
+				"id": data.Id,
+			})
+			resp.State.RemoveResource(ctx)
+			return
+		}
+	}
+}
+
+func (r *machineUserResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data machineUserResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update API call logic
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
```

seems like we don't do anything here. is this a supported operation just the operation logic is missing or we don't support updates and we should throw an error in case of calling this?